tests/check-refs.sh

```diff
@@ -40,3 +40,7 @@ nix-build -o $RESULT check-refs.nix -A test7
 
 # test10 should succeed (no disallowed references).
 nix-build -o $RESULT check-refs.nix -A test10
+
+# test11 should succeed (no referenceable paths)
+test11=$(nix-build -o $RESULT check-refs.nix -A test11)
+! nix-store -q --references "$test11" | grep -q .
```

Comment:
You probably want to use [isDaemonNewer](https://github.com/NixOS/nix/blob/cacfc32ee22bacb6400796e47fd4dee41299b66a/tests/common.sh.in#L133-L138) to only test this if the daemon is recent-enough. Maybe something like:

```suggestion
if isDaemonNewer 2.12pre20221001; then
  # test11 should succeed (no referenceable paths)
  test11=$(nix-build -o $RESULT check-refs.nix -A test11)
  ! nix-store -q --references "$test11" | grep -q .
fi
```